cnxarchive/database.py

```diff
@@ -186,9 +196,14 @@ def get_collections(module_ident, plpy):
         FROM trees c JOIN t ON (c.nodeid = t.parent)
         WHERE not c.nodeid = ANY(t.path)
     )
-    SELECT DISTINCT m.module_ident
-    FROM t JOIN latest_modules m ON (t.document = m.module_ident)
-    WHERE t.parent IS NULL ORDER BY m.module_ident
+    SELECT * FROM (
+        SELECT DISTINCT m.module_ident, m.revised,
+            MAX(m.revised) OVER (PARTITION BY m.uuid) as latest
+        FROM t, modules m
+        WHERE t.parent IS NULL and t.document = m.module_ident
+    ) r
+    WHERE r.revised = r.latest
+    ORDER BY r.module_ident
```

Comment:
This is not giving the same results as joining against latest_modules (on a quiescent database - no baking going on), when using a non-latest page module_ident. Not sure if that's really an issue, since this code should only ever be called  when a page has been published, and in that case the existing page must, _ipso facto_, be the current latest for that page.